src/class.jl

```diff
@@ -0,0 +1,63 @@
+"""
+    Class
+
+A type used for controlling which classes of `TimeZone` are considered valid. Instances of
+`Class` can be combined using bitwise operators to generate a mask which allows multiple
+classes to be considered valid at once.
+
+Currently supported masks are:
+
+- `Class.FIXED`: Class indicating the time zone name is parsable as a fixed UTC offset.
+- `Class.STANDARD`: The time zone name is included in the primary IANA tz source files.
+- `Class.LEGACY`: The time zone name is included in the deprecated IANA tz source files.
+- `Class.NONE`: Mask that will match nothing.
+- `Class.DEFAULT`: Default mask used by functions: `Class.FIXED | Class.STANDARD`
+- `Class.ALL`: Mask allowing all supported classes.
+"""
+struct Class
+    val::UInt8
+end
+
+function Base.getproperty(::Type{Class}, field::Symbol)
```

Comment:
Seems like a use case for `Enum`s?

Reply:
Doesn't work with flags and `|`

Reply:
It does if you define a `|` method

Reply:
Even if you define `|` you can't get any arbitrary combination of values:
```julia
julia> @enum Class NONE=0x00 FIXED=0x01 STANDARD=0x02 LEGACY=0x04

julia> Base.:(|)(a::Class, b::Class) = Class(Int(a) | Int(b))

julia> STANDARD | LEGACY
ERROR: ArgumentError: invalid value for Enum Class: 6
```

Reply:
FWIW, this `Class` approach seems fine to me. It's pretty close to what I did with file modes in FilePaths.jl:

https://github.com/rofinn/FilePathsBase.jl/blob/master/src/mode.jl